src/coreclr/jit/emitarm64.cpp

```diff
@@ -6993,13 +6993,6 @@ void emitter::emitIns_R_R_R_Ext(instruction ins,
         shiftAmount = insOptsLSL(opt) ? scale : 0;
     }
 
-    // If target reg is ZR - it means we're doing an implicit nullcheck
-    // where target type was ignored and set to TYP_INT.
-    if ((reg1 == REG_ZR) && (shiftAmount > 0))
-    {
-        shiftAmount = scale;
-    }
```

Comment:
this was added by me, it turns out it was masking the bug that existed since .NET Framework I guess (changed shiftAmount to bypass assert)